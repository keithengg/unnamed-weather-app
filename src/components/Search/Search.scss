.search {
    &__container {
        position: relative;
        display: flex;
        flex-direction: column;
        flex-grow: 1;

        margin-right: 16px;

        &:focus-within label {
            transform: translate(4px, 12px) scale(0.6);
            color: #0a53e4;
        }
    }

    &__input {
        width: 100%;
        height: 64px;

        background-color: rgba(355, 355, 355, 0.3);
        border-radius: 15px;
        padding: 20px;
    }

    &__label {
        position: absolute;
        color: #647493;
        font-size: 16px;
        line-height: 1;
        left: 16px;

        pointer-events: none;
        transform: translate(0, 23px) scale(1);
        transform-origin: top left;
        transition: 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;

        &--filled {
            transform: translate(4px, 12px) scale(0.6);
        }
    }

    &__button {
        display: flex;
        align-items: center;
        justify-content: center;
        height: 64px;
        width: 64px;

        background-color: rgba(116, 33, 138, 0.8) !important;
        border-radius: 15px;
        padding: 8px;
    }
}