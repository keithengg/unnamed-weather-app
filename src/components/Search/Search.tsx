import React from "react"

interface Props {
    location: string;
    handleUpdate: (location: string) => void;
    handleClick: () => void;
};

const Search: React.FC<Props> = (props) => {
    return (
        <>
            <label htmlFor="location">Enter Country/City</label>
            <input
                type="text"
                name="location"
                value={props.location}
                onChange={(e) => props.handleUpdate(e.target.value)}
            />
            <button onClick={props.handleClick} type="button" aria-label="Search">Search</button>
        </>
    )
}

export default Search;