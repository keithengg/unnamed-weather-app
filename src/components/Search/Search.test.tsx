import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom'
import Search from "./Search";

describe("<Search />", () => {
    const handleUpdate = jest.fn();
    const handleClick = jest.fn();

    it('should render correctly and match snapshot', () => {
        const { container } = render(
            <Search
                location=""
                handleClick={handleClick}
                handleUpdate={handleUpdate}
            />
        )
        expect(container).toMatchSnapshot()
    });

    it("should call handleClick when Search button clicked", async () => {
        render(
            <Search
                location="Singapore"
                handleClick={handleClick}
                handleUpdate={handleUpdate}
            />
        )

        const input = await screen.getByRole("textbox");
        expect((input as HTMLInputElement).value).toBe("Singapore")

        const button = screen.getByRole("button", { name: 'Search' });
        expect(button).toBeInTheDocument()

        fireEvent.click(button)
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});