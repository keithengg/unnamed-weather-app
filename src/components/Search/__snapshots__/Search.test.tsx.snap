// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly and match snapshot 1`] = `
<div>
  <label
    for="location"
  >
    Enter Country/City
  </label>
  <input
    name="location"
    type="text"
    value=""
  />
  <button
    aria-label="Search"
    type="button"
  >
    Search
  </button>
</div>
`;
